src/main/java/nl/ordina/jobcrawler/controller/JobcrawlerController.java

```diff
@@ -0,0 +1,80 @@
+package nl.ordina.jobcrawler.controller;
+
+import nl.ordina.jobcrawler.SearchRequest;
+import nl.ordina.jobcrawler.SearchResult;
+import nl.ordina.jobcrawler.model.Vacancy;
+import nl.ordina.jobcrawler.controller.exception.VacancyNotFoundException;
+import nl.ordina.jobcrawler.service.VacancyService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.*;
+
+
+@CrossOrigin(origins = "http://localhost:4200")
+@RestController
+public class JobcrawlerController {
+
+    @Autowired
+    private VacancyService vacancyService;
```

Comment:
Bij Alphabet werd ons verplicht om dependency injection (wat @AutoWired doet) via de constructor te regelen. Dit heeft meerdere voordelen. Onder andere dat je ervoor kunt zorgen dat het object altijd op de juiste manier aangemaakt wordt: door de dependencies in de constructor te checken, kun je een fout opgooien als er iets mis is.

Zie http://dillonbuchanan.com/programming/dependency-injection-constructor-vs-property/

Dit is overigens niet fout.

Reply:
Gewijzigd in https://github.com/mostafa-221/jobcrawler-backend/commit/a4770df2140e65eea0910c84c348653c2f8e4c2f